dbms/src/Storages/KVStore/KVStore.cpp

```diff
@@ -51,6 +51,35 @@ extern const char force_fail_in_flush_region_data[];
 extern const char pause_passive_flush_before_persist_region[];
 } // namespace FailPoints
 
+void PreHandlingTrace::waitForSubtaskResources(uint64_t region_id, size_t parallel)
+{
+    auto parallel_subtask_limit = kvstore->getMaxParallelPrehandleSize();
+    if (ongoing_prehandle_subtask_count.load() + parallel <= parallel_subtask_limit)
```

Comment:
Yes, but this rarely happens. Even if it does happen, no significant harm, because the count of vCores is limited.
However, a simple cas is more neat here, so I will fix that.

Reply:
Yes, we should only keep the logic after `cpu_resource_cv` wait returns

Reply:
fixed